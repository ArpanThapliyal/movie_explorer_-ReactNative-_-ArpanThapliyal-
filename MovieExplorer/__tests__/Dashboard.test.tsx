// __tests__/Dashboard.test.tsx
/**
 * @format
 */

import React from 'react';
import { render } from '@testing-library/react-native';

// 1) Stub out the dashboardHeader and dashboardBody components
jest.mock(
  '../src/components/dashboard/dashboardHeader',
  () => () => <></>
);
jest.mock(
  '../src/components/dashboard/dashboardBody',
  () => () => <></>
);

// Now import Dashboard
import Dashboard from '../src/screen/Dashboard';

describe('Dashboard screen', () => {
  it('renders without crashing', () => {
    const { toJSON } = render(<Dashboard />);
    expect(toJSON()).toBeTruthy();
  });

  it('matches snapshot structure', () => {
    const { toJSON } = render(<Dashboard />);
    expect(toJSON()).toMatchSnapshot();
  });
});
