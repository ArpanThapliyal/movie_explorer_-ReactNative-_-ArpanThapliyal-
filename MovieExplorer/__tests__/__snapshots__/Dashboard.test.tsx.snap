// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard screen matches snapshot structure 1`] = `
<View
  style={
    {
      "backgroundColor": "black",
      "flex": 1,
      "paddingTop": 40.019999999999996,
    }
  }
>
  <RCTScrollView
    keyboardShouldPersistTaps="handled"
    showsVerticalScrollIndicator={false}
  >
    <View />
  </RCTScrollView>
</View>
`;
